tools/fog-local-network/fog_conformance_tests.py

```diff
@@ -279,17 +283,19 @@ def balance_check(self, step_name, acceptable_answers_per_wallet):
         assert len(acceptable_answers_per_wallet) == self.NUM_WALLETS
 
         print(f'{step_name}: Performing fresh balance checks')
-        new_wallets = [
-            self._fresh_balance_check(step_name, wallet_num, acceptable_balances, expected_eventual_block_count)
-            for wallet_num, (acceptable_balances, expected_eventual_block_count) in enumerate(acceptable_answers_per_wallet)
-        ]
+        # The lambda we will pass to thread-pool
+        # input should be: (wallet_num, (acceptable_answers, expected_eventual_block_count))
+        map_fn = lambda args: self._fresh_balance_check(step_name, args[0], args[1][0], args[1][1])
+        new_wallets = self.thread_pool.map(map_fn, enumerate(acceptable_answers_per_wallet))
 
         if self.steps and not self.skip_followup_balance_checks:
-            print(f'{step_name}: Performing followup balance checks')
-            for wallets in self.steps:
-                for wallet, (acceptable_balances, expected_eventual_block_count) in zip(wallets, acceptable_answers_per_wallet):
-                    print(f'{step_name}: Followup balane check on {wallet.name} {wallet.key_num}...')
-                    self._follow_up_balance_check(wallet, acceptable_balances, expected_eventual_block_count)
+            # The lambda we will pass to thread-pool
+            #    print(f'{step_name}: Follow up balance check on {wallet.name} {wallet.key_num}...')
+            map_fn = lambda args: self._follow_up_balance_check(args[0], args[1][0], args[1][1])
+
+            print(f'{step_name}: Performing follow up balance checks')
+            iterables = [zip(wallets, acceptable_answers_per_wallet) for wallets in self.steps]
```

Comment:
I would add a comment on what this does, since I found it not obvious when first reading it.